### PR TITLE
DE575629 - Most program has errors in SQL-ERROR-RTN paragraph

### PR DESCRIPTION
**Pull request by ilidio-lopes**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/VariableUsageUtils.java [hunk 1]
```diff
@@ -68,6 +68,25 @@ public static List<VariableNode> findVariablesForUsage(
         : exactHierarchyMatchedVariables;
   }
 
+  private Collection<VariableNode> findDefinedVariable(String name, Multimap<String, VariableNode> definedVariables) {
+    Collection<VariableNode> foundVariable =  definedVariables.get(name);
+    if (foundVariable.size() > 0) {
+      return foundVariable;
+    }
+    outer:
+    for (VariableNode variableNode : definedVariables.values()) {
```
Comment (amanPras) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/VariableUsageUtils.java]: we could have used stream API 
```java
definedVariables.values()
            .stream().flatMap(Node::getDepthFirstStream)
            .filter(VariableNode.class::isInstance)
            .map(VariableNode.class::cast)
            .filter(var -> var.getName().equals(name))
            .findFirst(); // or .collect(Collectors.toList())
```

**Review by amanPras — CHANGES_REQUESTED** [review]
I suggest to add a UseCase test for this

**Comment by ishche** [comment]
Please update the PR description and squash commits.

**Comment by ilidio-lopes** [comment]
> Please update the PR description and squash commits.

I cannot squash. Probably because I don't have write permission.

**Review by amanPras — APPROVED** [review]
LGTM

**Review by ishche — APPROVED** [review]

**Pull request merged**